bowling.js

```diff
@@ -0,0 +1,36 @@
+class Bowling {
+  constructor() {
+    this.rolls = [];
+  }
+
+  roll(pins) {
+    this.rolls.push(pins);
+  }
+
+  score() {
+    let total = 0;
+    let turn = 0;
+    for(let i = 0 ; i < 10 ; i++) {
+      const frameScore = this.rolls[turn] + this.rolls[turn + 1];
+      if (this.rolls[turn] === 10) {
+        total += this.spareOrStrikeBonus(turn);
+        turn++;
+      } else if (frameScore === 10) {
+        total += this.spareOrStrikeBonus(turn);
+        turn += 2;
+      } else {
+        total += frameScore;
+        turn += 2;
+      }
+    }
+    return total;
+  }
```

Comment:
This was really cool! Took me a few minutes to understand it, I think combining strike and spare bonuses into one was super neat 👌